tests/dynamic_data/base_test.py

```diff
@@ -91,6 +91,12 @@ def parametrize_test_cases(cls) -> [TestCaseType]:
 
         test_cases = []
         for input_test_case in cls.test_cases:
+            specific_types = {}
+            for source_filter, test_type in input_test_case.specific_types.items():
+                filter_set = util.case_to_simple_parametrizations(source_filter)
+                for filter in filter_set:
+                    assert filter not in specific_types
+                    specific_types[filter] = test_type
```

Comment:
Encapsulate it to method and use the method for specific_types and specific_params.

Reply:
Also add unit tests (positive and negative)